### DISCUSSION TITLE
大佬，路由器里Docker怎么部署的

### DISCUSSION THREAD

**Discussion by Lee888888888888** [vvbbnn00/WARP-Clash-API#95] [Q&A]
如题[Screenshot_2024-02-24-16-06-05-109_com.android.chrome-edit.jpg](https://github.com/vvbbnn00/WARP-Clash-API/assets/62000336/6cfce9e6-6d52-4431-88d5-64fc2f66ad61)

**Comment by vvbbnn00** [comment]
你好，可以参照：https://github.com/vvbbnn00/WARP-Clash-API/issues/47#issuecomment-1960029271

### 示例

```bash
docker run -p 21001:3000 -e SECRET_KEY=123456 -v ./account:/app/account -v ./logs:/app/logs vvbbnn00/warp-clash-api:latest
```

### 解析

镜像：`vvbbnn00/warp-clash-api:latest`

端口映射：`21001`（外部） -> `3000`（内部）

环境变量参考：https://github.com/vvbbnn00/WARP-Clash-API?tab=readme-ov-file#-%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F
此处设置`SECRET_KEY`为`123456`

挂载：
- `./account:/app/account`
- `./logs:/app/logs`

**Comment by Lee888888888888** [comment]
> > ./
> 
> 似乎不能在 ARM 架构的软路由上面用

下载是没问题，编译项目的时候显示no configuration file provided: not found，Compose有，也显示版本了。可能是缺少其他环境的问题，你说的架构不支持也有可能，我是小白，瞎搞了一天也没搞好，就放弃了，还是用别人的成品算了😄

**Comment by jjjyyyllll** [comment]
似乎不能arm架构

The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /bin/sh: exec format error

**Comment by vvbbnn00** [comment]
> 似乎不能arm架构
> 
> The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested exec /bin/sh: exec format error

是从docker hub拉去的镜像吗

**Comment by jjjyyyllll** [comment]
> > 似乎不能arm架构
> > The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested exec /bin/sh: exec format error
> 
> 是从docker hub拉去的镜像吗

对啊 用本地软路由和arm64的服务器都试过 都是这个。  不支持ARM V8架构

**Comment by vvbbnn00** [comment]
@jjjyyyllll

docker hub里面之前只编译了amd64架构的，刚刚编译了一个测试用的arm架构镜像：

`vvbbnn00/warp-clash-api:sha-f9eaac5`

可以试试看能否运行。

**Comment by jjjyyyllll** [comment]
> @jjjyyyllll
> 
> docker hub里面之前只编译了amd64架构的，刚刚编译了一个测试用的arm架构镜像：
> 
> `vvbbnn00/warp-clash-api:sha-f9eaac5`
> 
> 可以试试看能否运行。

本地软路由可以了。 默认的环境变量，优选过IP后是不变的吧，感觉内存占用还挺高的？
![QQ拼音截图20240225171557](https://github.com/vvbbnn00/WARP-Clash-API/assets/3983163/3aaa9585-7092-47a6-a7be-b24afb72a18e)

**Comment by vvbbnn00** [comment]
优选IP用于选择最适合当前网络环境的WARP接入点，和docker镜像的IP地址无关。
内存的话在我的电脑上运行是`88MB`左右，如果显示占用过高可能存在着一些其他问题，需要进一步诊断。

**Comment by jjjyyyllll** [comment]
> 优选IP用于选择最适合当前网络环境的WARP接入点，和docker镜像的IP地址无关。 内存的话在我的电脑上运行是`88MB`左右，如果显示占用过高可能存在着一些其他问题，需要进一步诊断。

我这种本地家宽网络每天重启一次，ip会变的话，最好是设置定时重新优选ip的吧

**Comment by vvbbnn00** [comment]
> > 优选IP用于选择最适合当前网络环境的WARP接入点，和docker镜像的IP地址无关。 内存的话在我的电脑上运行是`88MB`左右，如果显示占用过高可能存在着一些其他问题，需要进一步诊断。
> 
> 我这种本地家宽网络每天重启一次，ip会变的话，最好是设置定时重新优选ip的吧

可以设置定时优选，不过其实感觉应该和ip关系不大，主要还是与物理地址有关

**Comment by jjjyyyllll** [comment]
我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？

**Comment by vvbbnn00** [comment]
> 我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？

那个目前还有点问题，我需要再研究一下😂

> vvbbnn00/warp-clash-api:sha-f9eaac5

这个目前是可以使用的

**Comment by jjjyyyllll** [comment]
> > 我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？
> 
> 那个目前还有点问题，我需要再研究一下😂
> 
> > vvbbnn00/warp-clash-api:sha-f9eaac5
> 
> 这个目前是可以使用的

好的。期待下大佬的最新 latest

**Comment by vvbbnn00** [comment]
> > > 我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？
> > 
> > 
> > 那个目前还有点问题，我需要再研究一下😂
> > > vvbbnn00/warp-clash-api:sha-f9eaac5
> > 
> > 
> > 这个目前是可以使用的
> 
> 好的。期待下大佬的最新 latest

现在应该可以使用了

**Comment by jjjyyyllll** [comment]
> > > > 我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？
> > > 
> > > 
> > > 那个目前还有点问题，我需要再研究一下😂
> > > > vvbbnn00/warp-clash-api:sha-f9eaac5
> > > 
> > > 
> > > 这个目前是可以使用的
> > 
> > 
> > 好的。期待下大佬的最新 latest
> 
> 现在应该可以使用了

docker部署，从warp+获取的key（2000W GB）的不能 用下面命令重置吗？？

curl -X POST http://host:port/api/account/update_license -H "Content-Type: application/json" -d "{\"license_key\": \"your_license_key\"}"

提示 
{"code":403,"message":"Unauthorized"}

**Comment by vvbbnn00** [comment]
> > > > > 我看hub上面 大佬你3分钟前上传一次 是否拉最新的latest 就可以了？
> 
> > > > 
> 
> > > > 
> 
> > > > 那个目前还有点问题，我需要再研究一下😂
> 
> > > > > vvbbnn00/warp-clash-api:sha-f9eaac5
> 
> > > > 
> 
> > > > 
> 
> > > > 这个目前是可以使用的
> 
> > > 
> 
> > > 
> 
> > > 好的。期待下大佬的最新 latest
> 
> > 
> 
> > 现在应该可以使用了
> 
> 
> 
> docker部署，从warp+获取的key（2000W GB）的不能 用下面命令重置吗？？
> 
> 
> 
> curl -X POST http://host:port/api/account/update_license -H "Content-Type: application/json" -d "{\"license_key\": \"your_license_key\"}"
> 
> 
> 
> 提示 
> 
> {"code":403,"message":"Unauthorized"}

似乎是因为设置了SECRETKEY但是没有传入 可以在url后面增加 ?key=xxxx

**Comment by jjjyyyllll** [comment]
> curl -X POST http://host:port/api/account/update_license -H "Content-Type: application/json" -d "{"license_key": "your_license_key"}"

还是报错。。
{"code":403,"message":"Unauthorized"}
curl: (3) URL using bad/illegal format or missing URL

**Comment by vvbbnn00** [comment]
curl -X POST http://host:port/api/account/update_license?key=xxxxx -H "Content-Type: application/json" -d "{\"license_key\": \"your_license_key\"}"

**Comment by vvbbnn00** [comment]
手机版github有些问题 会把转义符号给吞掉
具体参照readme中的写法：
![image](https://github.com/vvbbnn00/WARP-Clash-API/assets/46409975/8206289b-f079-498a-b819-c55fa525181c)

**Comment by jjjyyyllll** [comment]
> 手机版github有些问题 会把转义符号给吞掉 具体参照readme中的写法： ![image](https://private-user-images.githubusercontent.com/46409975/307583970-8206289b-f079-498a-b819-c55fa525181c.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NTgxMzAsIm5iZiI6MTcwODg1NzgzMCwicGF0aCI6Ii80NjQwOTk3NS8zMDc1ODM5NzAtODIwNjI4OWItZjA3OS00OThhLWI4MTktYzU1ZmE1MjUxODFjLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNVQxMDQzNTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OGQxZDcxYWJlNWM4YTAwNzUyOTk2MDhiMzVjZTJiOGZkYWY0ZjVlOWIyZjBkOGRkZWNkMDA0ZmNmNzFmNzIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Q8SuLo90y2dKsxpKBuxZB8LSy-6at8DE7AYWtSY10Tg)

感谢大佬不辞辛苦解惑，docker配置key后，如何重新获取订阅？ 我看没有重置按钮  以及 重启了docker 流量还是没变？

**Comment by vvbbnn00** [comment]
在手机或者电脑端的clash之类软件上重新下载一下订阅文件即可，剩余流量应该是显示有延迟，等两分钟左右就会更新的

**Comment by jjjyyyllll** [comment]
> 在手机或者电脑端的clash之类软件上重新下载一下订阅文件即可，剩余流量应该是显示有延迟，等两分钟左右就会更新的

非常感谢解惑！
